modules/base/rendering/renderablemodel.cpp

```diff
@@ -279,103 +282,173 @@ RenderableModel::RenderableModel(const ghoul::Dictionary& dictionary)
+    if (p.enableAnimation.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Attempting to enable animation for a model that does not have any");
+        }
+        else if (*p.enableAnimation &&_animationStart == "") {
+            LWARNING("Cannot enable animation without a given start time");
+        }
+        else {
+            _enableAnimation = *p.enableAnimation;
+            _geometry->enableAnimation(_enableAnimation.value());
+        }
+    }
+
+    if (p.animationTimeScale.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Animation time scale given to model without animation");
+        }
+        else if (std::holds_alternative<float>(*p.animationTimeScale)) {
+            _geometry->setTimeScale(std::get<float>(*p.animationTimeScale));
+        }
+        else if (std::holds_alternative<Parameters::AnimationTimeUnit>(*p.animationTimeScale)) {
+            Parameters::AnimationTimeUnit animationTimeUnit =
+                std::get<Parameters::AnimationTimeUnit>(*p.animationTimeScale);
+            TimeUnit timeUnit;
+
+            switch (animationTimeUnit) {
+                case Parameters::AnimationTimeUnit::Nanosecond:
+                    timeUnit = TimeUnit::Nanosecond;
+                    break;
+                case Parameters::AnimationTimeUnit::Microsecond:
+                    timeUnit = TimeUnit::Microsecond;
+                    break;
+                case Parameters::AnimationTimeUnit::Millisecond:
+                    timeUnit = TimeUnit::Millisecond;
+                    break;
+                case Parameters::AnimationTimeUnit::Second:
+                    timeUnit = TimeUnit::Second;
+                    break;
+                case Parameters::AnimationTimeUnit::Minute:
+                    timeUnit = TimeUnit::Minute;
+                    break;
+                default:
+                    throw ghoul::MissingCaseException();
             }
+
+            _geometry->setTimeScale(convertTime(1.0, timeUnit, TimeUnit::Second));
+        }
+    }
+
+    if (p.animationMode.has_value()) {
+        if (!_geometry->hasAnimation()) {
+            LWARNING("Animation mode given to model without animation");
+        }
+
+        switch (*p.animationMode) {
+            case Parameters::AnimationMode::LoopFromStart:
+                _animationMode = AnimationMode::LoopFromStart;
+                break;
+            case Parameters::AnimationMode::LoopInfinitely:
+                _animationMode = AnimationMode::LoopInfinitely;
+                break;
+            case Parameters::AnimationMode::BounceFromStart:
+                _animationMode = AnimationMode::BounceFromStart;
+                break;
+            case Parameters::AnimationMode::BounceInfinitely:
+                _animationMode = AnimationMode::BounceInfinitely;
+                break;
+            case Parameters::AnimationMode::Once:
+            default:
```

Comment:
Maybe better to move the code actually into the `Once` case and throw a `ghoul::MissingCaseException` here in the `default` label

Reply:
Otherwise, there is the danger of someone adding a new case and then needs to figure out where the missing label is, rather than getting a proper crash